src/main/java/io/airlift/units/Count.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.units;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static io.airlift.units.Preconditions.checkArgument;
+import static java.lang.Long.parseLong;
+import static java.lang.Math.floor;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class Count
+        implements Comparable<Count>
+{
+    private static final Pattern PATTERN = Pattern.compile("^\\s*(\\d+)\\s*([a-zA-Z]?)\\s*$");
+
+    // We iterate over the MAGNITUDES constant in convertToMostSuccinctRounded()
+    // instead of Magnitude.values() as the latter results in non-trivial amount of memory
+    // allocation when that method is called in a tight loop. The reason is that the values()
+    // call allocates a new array at each call.
+    private static final Magnitude[] MAGNITUDES = Magnitude.values();
+
+    /**
+     * @return count with no bigger value than 1000 in succinct magnitude, fractional part is rounded
+     */
+    public static Count succinctRounded(long count)
+    {
+        return succinctRounded(count, Magnitude.SINGLE);
+    }
+
+    /**
+     * @return count with no bigger value than 1000 in succinct magnitude, fractional part is rounded
```

Comment:
How is there a fractional part when `value` is a `long`?

Reply:
When you pass 123456 in SINGLE then you will get 123.456K that will be rounded to 123K.